managed/services/victoriametrics/prometheus.go

```diff
@@ -209,9 +217,68 @@ func AddInternalServicesToScrape(cfg *config.Config, s models.MetricsResolutions
 		scrapeConfigForAlertmanager(s.MR),
 		scrapeConfigForGrafana(s.MR),
 		scrapeConfigForPMMManaged(s.MR),
-		scrapeConfigForQANAPI2(s.MR))
+		scrapeConfigForQANAPI2(s.MR),
+		scrapeConfigForServerPMMAgent(s.MR))
 	// TODO Refactor to remove boolean positional parameter when Prometheus is removed
 	if dbaas {
 		cfg.ScrapeConfigs = append(cfg.ScrapeConfigs, scrapeConfigForDBaaSController(s.MR))
 	}
 }
+
+// AddPmmAgentScrapeConfig adds pmm-agent metrics to scrape targets.
+func AddPmmAgentScrapeConfig(l *logrus.Entry, q *reform.Querier, s *models.MetricsResolutions, pmmAgentID *string) (*config.ScrapeConfig, error) {
+	agent, err := models.FindAgentByID(q, pointer.GetString(pmmAgentID))
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	// sanity check
+	if (agent.NodeID != nil) && (agent.ServiceID != nil) {
+		l.Panicf("Both agent.NodeID and agent.ServiceID are present: %s", agent)
+	}
+
+	// find Service for this Agent
+	var paramsService *models.Service
+	if agent.ServiceID != nil {
+		paramsService, err = models.FindServiceByID(q, pointer.GetString(agent.ServiceID))
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
serviceID and NodeID are always nil for pmm-agent, so we can remove everything related to ServiceID and NodeID. it also means that paramsNode is always nil.